pkg/core/resources/apis/mesh/dataplane_helpers.go

```diff
@@ -179,6 +179,28 @@ func (d *DataplaneResource) IsIPv6() bool {
 	return ip.To4() == nil
 }
 
+func (d *DataplaneResource) IsUsingTransparentProxy() bool {
+	if d == nil {
+		return false
+	}
+	if d.Spec.GetNetworking() == nil {
+		return false
+	}
+	isUsingTransparentProxy := false
+	if d.Spec.GetNetworking().GetTransparentProxying() != nil &&
+		d.Spec.GetNetworking().GetTransparentProxying().RedirectPortInbound != 0 &&
+		d.Spec.GetNetworking().GetTransparentProxying().RedirectPortOutbound != 0 {
+		isUsingTransparentProxy = true
+	}
+	if d.IsIPv6() {
+		if d.Spec.GetNetworking().GetTransparentProxying() == nil ||
+			d.Spec.GetNetworking().GetTransparentProxying().RedirectPortInboundV6 == 0 {
+			isUsingTransparentProxy = false
+		}
+	}
+	return isUsingTransparentProxy
```

Comment:
wdyt?
```suggestion
	tproxy := d.Spec.GetNetworking().GetTransparentProxying()
	if tproxy == nil {
		return false
	}
	
	isUsingTransparentProxy := tproxy.RedirectPortInbound != 0 && tproxy.RedirectPortOutbound != 0
	
	if d.IsIPv6() {
		isUsingTransparentProxy = tproxy.RedirectPortInboundV6 != 0
	}
	
	return isUsingTransparentProxy
```
I'm kinda courious what in case where there is ipv6 configured and port is set, but there are no ports for tproxy for ipv4, how do we know that tproxy should be configured only for ipv6, and not for both ipv4 and ipv6 and it's just misconfigured?

Reply:
I think we do not say anywhere in the docs to set only ipv6 but looking at the available configuration I wanted to check both. In this case, I am checking if the address defined in networking is IPv6.